performance-ptbr.md

```diff
@@ -181,6 +181,83 @@ Idéias para melhorar sua estrutura de dados:
 
 * Se as consultas forem caras, adicione um cache.
 
-  Em um nível maior, um cache interno ou externo (como o memcache) pode ajudar.
-  Pode ser excessivo para uma única estrutura de dados, nós vamos cobrir mais sobre caches abaixo.
+  Em um nível maior, um cache interno ou externo (como o memcache) pode ajudar. Isso pode soar excessivo para somente uma estrutura de dados. Falaremos mais sobre cache abaixo.
 
+Esses tipos de alterações são utéis quando os dados necessários são baratos para armanezar e fáceis de se manterem atualizados.
+
+Estes são exemplos claros de "Tenha menos trabalho" pensando no nível de estrutura de dados. Todos eles custam espaço. Na maior parte do tempo se você está otimizando pensando em uso de CPU, seu programa usará mais memório este é a clássica [compensação espaço-tempora](https://en.wikipedia.org/wiki/Space%E2%80%93time_tradeoff).
+
+É importante pensar como essa compensação pode afetar as suas soluções -- de maneira não direta. Às vezes, uma pequena quantidade de memória pode resultar em uma melhoria significativa de velocidade, em outras situações este tradeoff é linear (2x o uso da memória == 2x a melhora de desempenho), em outras vezes é significativamente pior: uma enorme quantidade de memória fornece apenas uma pequena melhora de desempenho. Onde você precisa estar nesta curva de memória/desempenho  pode afetar quais opções de algoritmos são razoáveis. Nem sempre é possível somente ajustar um parâmetro de um algoritmo. Diferentes usos de memória pode ter abordagens algorítimicas completamente diferentes.
+ 
+
+Tabelas de pesquisa também se enquadram nessa compensação espaço-temporal. Uma tabela de pesquisa simples
```

Comment:
Remove `\n` splitting sentences (there are many further down the text). Those linebreaks make the review of the markdown harder.

Reply:
OK!